Resources/doc/usage.rst

```diff
@@ -0,0 +1,180 @@
+            options:
+                expose: true
+
+        my_route_to_expose_with_defaults:
+            pattern: /blog/{page}
+            defaults: { _controller: AcmeBlogBundle:Blog:index, page: 1 }
+            options:
+                expose: true
+
+You can use the ``generate()`` method that way:
+
+.. code-block:: javascript
+
+    Routing.generate('my_route_to_expose', { id: 10 });
+    // will result in /foo/10/bar
+
+    Routing.generate('my_route_to_expose', { id: 10, foo: "bar" });
+    // will result in /foo/10/bar?foo=bar
+
+    $.get(Routing.generate('my_route_to_expose', { id: 10, foo: "bar" }));
+    // will call /foo/10/bar?foo=bar
+
+    Routing.generate('my_route_to_expose_with_defaults');
+    // will result in /blog/1
+
+    Routing.generate('my_route_to_expose_with_defaults', { id: 2 });
+    // will result in /blog/2
+
+    Routing.generate('my_route_to_expose_with_defaults', { foo: "bar" });
+    // will result in /blog/1?foo=bar
+
+    Routing.generate('my_route_to_expose_with_defaults', { id: 2, foo: "bar" });
+    // will result in /blog/2?foo=bar
+
+Moreover, you can configure a list of routes to expose in ``app/config/config.yml``:
+
+.. code-block:: yaml
+
+    # app/config/config.yml
+    fos_js_routing:
+        routes_to_expose: [ route_1, route_2, ... ]
+
+These routes will be added to the exposed routes. You can use regular expression
+patterns if you don't want to list all your routes name by name.
+
+You can prevent to expose a route by configuring it as below:
+
+.. code-block:: yaml
+
+    # app/config/routing.yml
+    my_very_secret_route:
+        pattern: /admin
+        defaults: { _controller: HelloBundle:Admin:index }
+        options:
+            expose: false
+
+HTTP Caching
+------------
+
+You can enable HTTP caching as below:
+
+.. code-block:: yaml
+
+    # app/config/config.yml
+    fos_js_routing:
+        cache_control:
+            # All are optional, defaults shown
+            public: false   # can be true (public) or false (private)
+            maxage: null    # integer value, e.g. 300
+            smaxage: null   # integer value, e.g. 300
+            expires: null   # anything that can be fed to "new \DateTime($expires)", e.g. "5 minutes"
+            vary: []        # string or array, e.g. "Cookie" or [ Cookie, Accept ]
+
+Commands
+--------
+
+fos:js-routing:dump
+~~~~~~~~~~~~~~~~~~~
+
+This command dumps the route information into a file so that instead of having
+the controller generated JavaScript, you can use a normal file. This also allows
+to combine the routes with the other JavaScript files in assetic.
+
+.. code-block:: bash
+
+    $ php app/console fos:js-routing:dump
+
+Instead of the line
+
+.. code-block:: twig
+
+    <script src="{{ path('fos_js_routing_js', {"callback": "fos.Router.setData"}) }}"></script>
+
+you now include this as
+
+.. code-block:: html
+
+    <script src="/js/fos_js_routes.js"></script>
```

Comment:
this should use the ``asset()`` function to load the JS file, to be safer (and to use the proper cache busting)